src/query/plan/operator.cpp

```diff
@@ -4565,30 +4572,64 @@ class CallProcedureCursor : public Cursor {
                                     self_->procedure_name_, get_proc_type_str(self_->is_write_),
                                     get_proc_type_str(proc->info.is_write));
       }
+      if (!proc->info.is_batched) {
+        stream_exhausted = true;
+      }
+
+      if (stream_exhausted) {
+        if (!input_cursor_->Pull(frame, context)) {
+          if (proc->cleanup) {
+            proc->cleanup.value()();
+          }
+          return false;
+        } else {
+          stream_exhausted = false;
+          if (proc->initializer) {
+            call_initializer = true;
+            MG_ASSERT(proc->cleanup);
+            proc->cleanup.value()();
+          }
+        }
+      }
+      if (!cleanup_ && proc->cleanup) [[unlikely]] {
+        cleanup_.emplace(*proc->cleanup);
+      }
+      // Unpluging memory without calling destruct on each object since everything was allocated with this memory
+      // resource
+      self_->monotonic_memory.Release();
+      result_ =
```

Comment:
Is the result_ completely destroyed by releasing the monotonic_memory?

Reply:
Yes, should be since it should be allocated on the heap.